--- asterisk-1.6.0-beta1-orig/main/config.c	2008-01-08 07:42:33.000000000 +1030
+++ asterisk-1.6.0-beta1/main/config.c	2008-01-22 14:50:12.000000000 +1030
@@ -49,7 +49,7 @@
 
 #include <glob.h>
 
-#ifdef SOLARIS
+#if defined(SOLARIS) || defined(__BLACKFIN__)
 #define MY_GLOB_FLAGS	GLOB_NOCHECK
 #else
 #define MY_GLOB_FLAGS	(GLOB_NOMAGIC|GLOB_BRACE)
--- asterisk-1.6.0-beta1-orig/main/manager.c	2008-01-19 08:34:33.000000000 +1030
+++ asterisk-1.6.0-beta1/main/manager.c	2008-01-22 14:50:12.000000000 +1030
@@ -2378,7 +2378,7 @@
 	const char *module = astman_get_header(m, "Module");
 	const char *id = astman_get_header(m,"ActionID");
 	char idText[BUFSIZ];
-	const char *version;
+	char *version = NULL;
 	char filename[BUFSIZ/2];
 	char *cut;
 
@@ -2397,8 +2397,9 @@
 	}
 	sprintf(cut, ".c");
 	ast_log(LOG_DEBUG, "**** ModuleCheck .c file %s\n", filename);
+#if !defined(LOW_MEMORY)
 	version = ast_file_version_find(filename);
-
+#endif
 	if (!ast_strlen_zero(id))
 		snprintf(idText, sizeof(idText), "ActionID: %s\r\n", id);
 	astman_append(s, "Response: Success\r\n%s", idText);
--- asterisk-1.6.0-beta1-orig/main/tcptls.c	2008-01-19 08:34:33.000000000 +1030
+++ asterisk-1.6.0-beta1/main/tcptls.c	2008-01-22 14:50:12.000000000 +1030
@@ -359,7 +359,9 @@
 void *ast_make_file_from_fd(void *data)
 {
 	struct server_instance *ser = data;
+#ifdef DO_SSL
 	int (*ssl_setup)(SSL *) = (ser->client) ? SSL_connect : SSL_accept;
+#endif
 	int ret;
 	char err[256];
 
--- asterisk-1.6.0-beta1-orig/res/ael/ael.flex	2008-01-11 08:16:56.000000000 +1030
+++ asterisk-1.6.0-beta1/res/ael/ael.flex	2008-01-22 14:50:12.000000000 +1030
@@ -418,7 +418,7 @@
 		   snprintf(fnamebuf2,sizeof(fnamebuf2), "%s/%s", (char *)ast_config_AST_CONFIG_DIR, fnamebuf);
 		   ast_copy_string(fnamebuf,fnamebuf2,sizeof(fnamebuf));
 		}
-#ifdef SOLARIS
+#if defined(SOLARIS) || defined (__BLACKFIN__) 
 			glob_ret = glob(fnamebuf, GLOB_NOCHECK, NULL, &globbuf);
 #else
 			glob_ret = glob(fnamebuf, GLOB_NOMAGIC|GLOB_BRACE, NULL, &globbuf);
--- asterisk-1.6.0-beta1-orig/res/ael/ael_lex.c	2008-01-11 08:16:56.000000000 +1030
+++ asterisk-1.6.0-beta1/res/ael/ael_lex.c	2008-01-22 14:50:12.000000000 +1030
@@ -1713,7 +1713,7 @@
 		   snprintf(fnamebuf2,sizeof(fnamebuf2), "%s/%s", (char *)ast_config_AST_CONFIG_DIR, fnamebuf);
 		   ast_copy_string(fnamebuf,fnamebuf2,sizeof(fnamebuf));
 		}
-#ifdef SOLARIS
+#if defined(SOLARIS) || defined(__BLACKFIN__)
 			glob_ret = glob(fnamebuf, GLOB_NOCHECK, NULL, &globbuf);
 #else
 			glob_ret = glob(fnamebuf, GLOB_NOMAGIC|GLOB_BRACE, NULL, &globbuf);
--- asterisk-1.6.0-beta1-orig/utils/Makefile	2007-12-19 09:36:05.000000000 +1030
+++ asterisk-1.6.0-beta1/utils/Makefile	2008-01-22 14:50:12.000000000 +1030
@@ -17,7 +17,7 @@
 .PHONY: clean all uninstall
 
 # to get check_expr, add it to the ALL_UTILS list
-ALL_UTILS:=astman smsq stereorize streamplayer aelparse muted check_expr conf2ael hashtest2 hashtest astcanary
+ALL_UTILS:=astman smsq stereorize streamplayer aelparse conf2ael astcanary
 UTILS:=$(ALL_UTILS)
 
 LIBS += $(BKTR_LIB)	# astobj2 with devmode uses backtrace
