diff -Naur asterisk-orig/main/loader.c asterisk/main/loader.c
--- asterisk-orig/main/loader.c	2007-06-07 08:17:11.000000000 -0400
+++ asterisk/main/loader.c	2007-06-07 08:34:03.000000000 -0400
@@ -530,6 +530,11 @@
 		ast_verbose("The previous reload command didn't finish yet\n");
 		return -1;	/* reload already in progress */
 	}
+	if (ast_lastreloadtime && time(NULL) - ast_lastreloadtime < 5) {
+                        ast_verbose("The previous reload was less than 5 seconds ago.\n");
+                        ast_mutex_unlock(&reloadlock);
+                        return -1;
+       }
 	ast_lastreloadtime = time(NULL);
 
 	/* Call "predefined" reload here first */
diff -Naur asterisk-orig/main/manager.c asterisk/main/manager.c
--- asterisk-orig/main/manager.c	2007-06-07 08:17:11.000000000 -0400
+++ asterisk/main/manager.c	2007-06-07 08:36:16.000000000 -0400
@@ -1041,7 +1041,7 @@
 		astman_send_error(s, m, "Filename not specified");
 		return 0;
 	}
-	if (!(cfg = ast_config_load_with_comments(fn))) {
+	if (!(cfg = ast_config_load(fn))) {
 		astman_send_error(s, m, "Config file not found");
 		return 0;
 	}
@@ -1151,7 +1151,7 @@
 		astman_send_error(s, m, "Filename not specified");
 		return 0;
 	}
-	if (!(cfg = ast_config_load_with_comments(sfn))) {
+	if (!(cfg = ast_config_load(sfn))) {
 		astman_send_error(s, m, "Config file not found");
 		return 0;
 	}
