--- zaptel-1.4.9.2-orig/Makefile	2008-05-19 10:29:22.000000000 -0400
+++ zaptel-1.4.9.2/Makefile		2008-05-19 10:32:19.000000000 -0400
@@ -93,8 +93,8 @@
 endif
 #SUBDIRS_UTILS	+= ppp
 
-TOPDIR_MODULES:=pciradio tor2 torisa wcfxo wct1xxp wctdm wcte11xp wcusb zaptel ztd-eth ztd-loc ztdummy ztdynamic zttranscode
-SUBDIR_MODULES:=wct4xxp wctc4xxp xpp wctdm24xxp wcte12xp
+TOPDIR_MODULES:=zaptel bfsi wcfxs sport_interface ztdummy
+SUBDIR_MODULES:=
 TOPDIR_MODULES+=$(MODULES_EXTRA)
 SUBDIR_MODULES+=$(SUBDIRS_EXTRA)
 BUILD_TOPDIR_MODULES:=$(filter-out $(MENUSELECT_MODULES),$(TOPDIR_MODULES))
@@ -136,7 +136,7 @@
 CFLAGS_PPC:=-fsigned-char
 endif
 ifneq (,$(findstring x86_64,$(UNAME_M)))
-CFLAGS_x86_64:=-m64
+#CFLAGS_x86_64:=-m64
 endif
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_x86_64)
 KFLAGS=-I$(KINCLUDES) -O6
